pkg/api/store/cluster/cluster_store.go

```diff
@@ -17,3 +26,29 @@ func (r *Store) ByID(apiContext *types.APIContext, schema *types.Schema, id stri
 	}
 	return r.Store.ByID(apiContext, schema, id)
 }
+
+func (r *Store) Create(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}) (map[string]interface{}, error) {
+
+	name, ok := data["name"].(string)
+	if !ok {
+		return nil, errors.New("Cluster name is required")
+	}
+
+	r.mu.Lock()
```

Comment:
this code should be invoked on the update as well

Reply:
Added the update method